### PR TITLE
clarify on deprecated content

### PR DESCRIPTION
**Pull request by alexisintech**
This PR adds `<Callout />` components with `type="danger"` (instead of `type="info"`) to better highlight that something is deprecated. This PR also adds "(deprecated)" copy where necessary (in places that deprecated items were mentioned).

This PR also updates practices in reflection of [these major updates](https://github.com/clerk/javascript/blob/main/packages/nextjs/CHANGELOG.md#500-alpha-v50):
- use NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY instead of NEXT_PUBLIC_CLERK_FRONTEND_API env variable
- use NEXT_PUBLIC_CLERK_JS_VERSION instead of CLERK_JS_VERSION
- use CLERK_SECRET_KEY instead of CLERK_API_KEY
- use publishableKey instead of frontendApi
- use @clerk/nextjs instead of @clerk/nextjs/ssr
- use @clerk/nextjs instead of @clerk/nextjs/api
- use middleware with authMiddleware instead of withClerkMiddleware

Fixes DOCS-499

### CODE REVIEW

File: docs/references/javascript/organization/members.mdx [hunk 1]
```diff
@@ -23,7 +23,7 @@ Retrieve the members of the currently active organization.
 | `pageSize?` | `number` | A number that indicates the maximum number of results that should be returned for a specific page. |
 | `role?` | `('admin' \| 'basic_member' \| 'guest_member')[]` | The roles of memberships that will be included in the response. |
 
-<Callout type="info">
+<Callout type="danger">
   `GetMembershipsParams` was recently updated to `GetMembersParams`. The `limit` and `offset` parameters are now deprecated.
```
Comment (rachelnabors) [inline docs/references/javascript/organization/members.mdx:27]: "recently" is an indefinite term. Can we provide a date?

File: docs/references/javascript/organization/organization.mdx [hunk 2]
```diff
@@ -14,7 +14,7 @@ This is a class that represents an organization. It's returned from the [`create
 | `id` | `string` | The unique identifier of the related organization. |
 | `name` | `string` | The name of the related organization. |
 | `slug` | `string \| null` | The organization slug. If supplied, it must be unique for the instance. |
-| `logoUrl` | `string \| null` | The full URL for the organization logo. (deprecated in favor of `imageUrl`) |
+| `logoUrl` (deprecated) | `string \| null` | The full URL for the organization logo. (deprecated in favor of `imageUrl`) |
```
Comment (rachelnabors) [inline docs/references/javascript/organization/organization.mdx:17]:
```suggestion
| `logoUrl` (deprecated) | `string \| null` | Deprecated in favor of `imageUrl`. |
```

File: docs/references/javascript/user/user.mdx [hunk 3]
```diff
@@ -24,7 +24,7 @@ The ClerkJS SDK provides some helper [methods](#methods) on the `User` object to
 | `username` | `string \| null` | The user's username. |
 | `imageUrl` | `string` | Holds the users profile image or avatar. |
 | `hasImage` | `boolean` | A getter boolean to check if the user has uploaded an image or one was copied from OAuth. Returns `false` if Clerk is displaying an avatar for the user. |
-| `profileImageUrl` | `string \| null` | The URL for the user's profile image.<br />**This property is deprecated** in favor of `image_url` |
+| `profileImageUrl` (deprecated) | `string \| null` | The URL for the user's profile image.<br />This property is deprecated in favor of `image_url`. |
```
Comment (rachelnabors) [inline docs/references/javascript/user/user.mdx:27]:
```suggestion
| `profileImageUrl` (deprecated) | `string \| null` | Deprecated in favor of `image_url`. |
```

File: docs/references/javascript/user/user.mdx [hunk 4]
```diff
@@ -71,7 +71,7 @@ All props below are optional.
 | Name | Type | Description |
 | --- | --- | --- |
 | `username` | `string` | The user's username. |
-| `password` | `string` | The user's password.<br />**This property is deprecated.** This will be removed in the next major version. Please use [`updatePassword(params)`](/docs/references/javascript/user/password-management#updatepassword) instead. |
+| `password` (deprecated) | `string` | The user's password.<br />This property is deprecated. This will be removed in the next major version. Please use [`updatePassword(params)`](/docs/references/javascript/user/password-management#update-password) instead. |
```
Comment (rachelnabors) [inline docs/references/javascript/user/user.mdx:74]: What # is the next version?

File: docs/references/nextjs/with-clerk-middleware.mdx [hunk 5]
```diff
@@ -5,7 +5,7 @@ description: The withClerkMiddleware wrapper allows Clerk to access session data
 
 # `withClerkMiddleware()`
 
-<Callout type="warning">
+<Callout type="danger">
   `withClerkMiddleware()` is Clerk's previous middleware API and will soon be deprecated. Please use [`authMiddleware()`](/docs/references/nextjs/auth-middleware) going forward.
```
Comment (rachelnabors) [inline docs/references/nextjs/with-clerk-middleware.mdx:9]: Can we give something more concrete than "soon?" A date, a release number, or just "an upcoming release"?

**Review by rachelnabors — CHANGES_REQUESTED** [review]
A vast improvement! (I would add a screenshot to this description of what the top of the pages look like for completeness, but not necessary at this stage.)

**Review by jescalan — APPROVED** [review]
🙌

**Comment by alexisintech** [comment]
> A vast improvement! (I would add a screenshot to this description of what the top of the pages look like for completeness, but not necessary at this stage.)

I have to disagree and say that I feel like screenshots of every different page that I made changes to are not a good use of time. There are preview deployments that can be explored by the reviewers.

**Review by JacobMGEvans — APPROVED** [review]

**Review by rachelnabors — APPROVED** [review]

**Pull request merged**